src/config.rs

```diff
@@ -126,6 +131,12 @@ impl Config {
 
             csp_report_only: env("DOCSRS_CSP_REPORT_ONLY", false)?,
 
+            local_archive_cache_path: env(
+                "DOCSRS_ARCHIVE_INDEX_CACHE_PATH",
+                PathBuf::from(".archive_cache"),
+            )?,
+            use_archive_storage: env("DOCSRS_USE_ARCHIVE_STORAGE", false)?,
```

Comment:
Hmm, is there a reason you made this configurable? I'd expect us to always do this unless it's completely broken, and if it's completely broken we should just revert instead of changing it at runtime.

Reply:
I think initially I added it as kind of feature-switch out of habit, the second usage came with the build-tests which set the setting to test both techniques. 

I'm happy to remove it if needed.

Reply:
Ahh, I missed that this is used in build-tests. Hmm, what I'd really like is a test that if we upload a file using the original format, then download it with `USE_ARCHIVE_STORAGE` set, then it still works properly. Is that what the build test is doing today? If not, then I don't think the flag is very useful.

Reply:
the second build-test is building with archive storage and then checking the storage for content and requesting some files. 

But thinking about this:
Assuming an eventual revert will be done without a flag, by just reverting the binary, 

then the builder only has to support archive-storage and nothing else any more. 

> Hmm, what I'd really like is a test that if we upload a file using the original format, then download it with USE_ARCHIVE_STORAGE set, then it still works properly 

isn't that (kind of) covered in the respective rustdoc/sources tests? we're testing everything with both possibilities.

Reply:
> isn't that (kind of) covered in the respective rustdoc/sources tests? we're testing everything with both possibilities.

Oh I see, you store whether the crate originally used `archive_storage` by adding a boolean to CrateDetails - only the upload depends on config, the download will still use whatever it was originally uploaded with.

Ok, in that case this makes sense to me, but I don't think we need an environment variable for it, just a test-only boolean on `Config`.

Reply:
@jyn514 I'm not sure I follow 100%. 

if we don't need the environment variable, we won't have production builds with the old storage after this is deployed, 
which means I can drop the whole variable and drop the old storage logic from the rustwide-builder. 

Accessing releases that were build without archive is handled in the respective tests with their fakes.

Reply:
I added a commit removing it, but I'm happy to discuss details

Reply:
> if we don't need the environment variable, we won't have production builds with the old storage after this is deployed,
which means I can drop the whole variable and drop the old storage logic from the rustwide-builder.

What? No, we definitely will, we'll still have all the builds that happened before this was deployed. We need to be able to retrieve those builds too, not just newly built crates.

Reply:
Ok, I was confused - in tests, this happens through fakes.rs, not through rustwide_builder. So you're correct, there's no need to keep the old code around.